### PR TITLE
[DOCS] Miscellaneous docstrings for `DataContext` and utils

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Docstring party

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code

### CODE REVIEW

File: great_expectations/cli/toolkit.py [hunk 1]
```diff
@@ -216,27 +216,26 @@ def load_expectation_suite(  # type: ignore[return] # sys.exit if no suite
     if expectation_suite_name.endswith(".json"):
         expectation_suite_name = expectation_suite_name[:-5]
 
-    suite: Optional[ExpectationSuite]
+    suite: Optional[ExpectationSuite] = None
```
Comment (cdkini) [inline great_expectations/cli/toolkit.py:219]: `mypy` randomly started complaining - small fix but should resolve the issue

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 2]
```diff
@@ -2433,29 +2477,24 @@ def add_batch_kwargs_generator(
         )
         return generator
 
+    @public_api
     def get_available_data_asset_names(
-        self, datasource_names=None, batch_kwargs_generator_names=None
+        self,
+        datasource_names: str | list[str] | None = None,
+        batch_kwargs_generator_names: str | list[str] | None = None,
```
Comment (cdkini) [inline great_expectations/data_context/data_context/abstract_data_context.py:2489]: Type annotating this caused resulted in some irrelevant type ignores

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 3]
```diff
@@ -2485,7 +2524,9 @@ def get_available_data_asset_names(
 
             elif len(batch_kwargs_generator_names) == 1:
                 datasource = self.get_datasource(datasource_names[0])
-                datasource_names[
+                # 20230120 - Chetan - I believe this is a latent bug - we should not be doing string-based indexing
+                #                     within a list. This will result in a runtime error.
+                datasource_names[  # type:ignore[call-overload]
```
Comment (cdkini) [inline great_expectations/data_context/data_context/abstract_data_context.py:2534]: Bug to be filed to investigate this - leaving it alone as it has been sitting around for some time and requires some domain knowledge to fully understand

Comment (Shinnnyshinshin) [reply great_expectations/data_context/data_context/abstract_data_context.py:2534]: Thank you for catching this @cdkini

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you very much for this @cdkini 🚀

**Pull request merged**